site/content/blog/2020-11-17-mysql-index-merge.md

```diff
@@ -0,0 +1,178 @@
+|  2 | SUBQUERY    | Posts | NULL       | const       | PRIMARY                                                                                                                              | PRIMARY                                  | 106     | const |      1 |   100.00 | NULL                                                                                   |
++----+-------------+-------+------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------+---------+-------+--------+----------+----------------------------------------------------------------------------------------+
+2 rows in set (0.00 sec)
+```
+
+This was strange. Instead of using an index, it was doing an `index_merge` of `idx_posts_channel_id` and `idx_posts_delete_at`. We also ran the old query just to be sure that nothing odd was happening:
+
+```
++----+-------------+-------+------------+-------+--------------------------------------------------------------------------------------------------------------------------------------+---------------------+---------+-------+---------+----------+----------------------------------+
+| id | select_type | table | partitions | type  | possible_keys                                                                                                                        | key                 | key_len | ref   | rows    | filtered | Extra                            |
++----+-------------+-------+------------+-------+--------------------------------------------------------------------------------------------------------------------------------------+---------------------+---------+-------+---------+----------+----------------------------------+
+|  1 | PRIMARY     | p     | NULL       | range | idx_posts_create_at,idx_posts_delete_at,idx_posts_channel_id,idx_posts_channel_id_update_at,idx_posts_channel_id_delete_at_create_at | idx_posts_create_at | 9       | NULL  | 6175639 |     1.58 | Using where; Backward index scan |
+|  2 | SUBQUERY    | Posts | NULL       | const | PRIMARY                                                                                                                              | PRIMARY             | 106     | const |       1 |   100.00 | NULL                             |
++----+-------------+-------+------------+-------+--------------------------------------------------------------------------------------------------------------------------------------+---------------------+---------+-------+---------+----------+----------------------------------+
+```
+
+As expected, the old query was selecting only `idx_posts_create_at`, but now the new query has decided to do something completely different. I wasn't too familiar with what was an index merge intersection, so I had to do some reading to understand what exactly was happening.
```

Comment:
```suggestion
As expected, the old query was selecting only `idx_posts_create_at`, but now the new query had decided to do something completely different. I wasn't too familiar with the index merge intersection, so I had to do some reading to understand what exactly was happening.
```

Reply:
@justinegeffen - This part "I wasn't too familiar with the index merge intersection" doesn't feel right. Does "I wasn't too familiar with what an index merge intersection was" feel better?